python/thunder/imgprocessing/transformation.py

```diff
@@ -0,0 +1,58 @@
+""" Transformations produced by registration methods """
+
+
+class Transformation(object):
+    """ Base class for transformations """
+
+    def apply(self, im):
+        raise NotImplementedError
+
+
+class Displacement(Transformation):
+
+    def __init__(self, delta=None):
+        self.delta = delta
+
+    def apply(self, im):
+        """
+        Apply an n-dimensional displacement by shifting an image or volume.
+
+        Parameters
+        ----------
+        im : ndarray
+            The image or volume to shift
+        """
+        from scipy.ndimage.interpolation import shift
+
+        return shift(im, map(lambda x: -x, self.delta), mode='nearest')
+
+    def __repr__(self):
+        return "Displacement(delta=%s)" % str(self.delta)
+
+
+class PlanarDisplacement(Transformation):
+
+    def __init__(self, delta=None):
```

Comment:
How is delta structured? Would be nice to document it here so you don't have to cross-reference the registration method to figure it out.